test/e2e/mcd_test.go

```diff
@@ -119,14 +119,20 @@ func TestKernelArguments(t *testing.T) {
 	infraNode := helpers.GetSingleNodeByRole(t, cs, "infra")
 	assert.Equal(t, infraNode.Annotations[constants.CurrentMachineConfigAnnotationKey], renderedConfig)
 	assert.Equal(t, infraNode.Annotations[constants.MachineConfigDaemonStateAnnotationKey], constants.MachineConfigDaemonStateDone)
-	kargs := helpers.ExecCmdOnNode(t, cs, infraNode, "cat", "/rootfs/proc/cmdline")
-	expectedKernelArgs := []string{"nosmt", "foo=bar", "foo=baz", "baz=test", "bar=hello world"}
-	for _, v := range expectedKernelArgs {
-		if !strings.Contains(kargs, v) {
-			t.Fatalf("Missing %q in kargs: %q", v, kargs)
+
+	// Non-CoreOS nodes don't support kernel/args today, we need to make sure they survive this test
+	if helpers.IsCoreOSNode(&infraNode) {
+		kargs := helpers.ExecCmdOnNode(t, cs, infraNode, "cat", "/rootfs/proc/cmdline")
+		expectedKernelArgs := []string{"nosmt", "foo=bar", "foo=baz", "baz=test", "bar=hello world"}
+		for _, v := range expectedKernelArgs {
+			if !strings.Contains(kargs, v) {
+				t.Fatalf("Missing %q in kargs: %q", v, kargs)
+			}
 		}
+		t.Logf("Node %s has expected kargs", infraNode.Name)
+	} else {
```

Comment:
this test change doesn't quite makes sense bc we are explicitly testing the above Fatalf error for a specific state so we don't need to be checking if it's a coreosnode or have the else as this test should _only_ pass if the fatal is not hit.

Reply:
Right, but while the test failure should be fatal for a CoreOS node, for a RHEL node it shouldn't. If you run our e2e tests against a cluster with RHEL nodes in it (like with make test-e2e), the e2e tests fail even though the MCO and the RHEL nodes are behaving properly -- RHEL nodes can receive kernel config during normal operation, but they're supposed to do nothing with it by design.
	
I was looking at this as "test failure" == "something is broken", and nothing is broken when the changes do not take effect on RHEL nodes.

 Is there some sort of general governing principle as to what goes in our repo's e2e tests and what needs to go in release/step-registry?

Reply:
why not just skip the test entirely if it can't be run on RHELX nodes? I'm not fully understanding the utility of having it roll out things that shouldn't work instead of just not running the test at all.

Reply:
if it's in a mixed node cluster (i have no idea how the suite running on rhel nodes is set up or if this exists), it selects a node to be a pool to roll a change out to, why not just check whether or not that node is RHELX there instead of rolling out the machineconfig that isn't supported?

Reply:
I can do it there, too, but I wasn't looking at the rest of the test being useless. I was looking at it as testing what happens in a mixed node cluster to make sure that you can: 

1.  "push out a config with unsupported features"
2. "do nothing with the parts you can't understand, like you're supposed to" 
3.  "make sure you can roll back from it" 

because you could receive unsupported config in a mixed-node cluster, and  you'd want to be sure it behaved properly? 

I actually tested it by pulling all of my CoreOS nodes out of my cluster and leaving only the RHEL ones so I wasn't at the mercy of the random chooser.

Reply:
> "do nothing with the parts you can't understand, like you're supposed to"

the problem is the there is no check in the else seeing if the node/mcd did what it was supposed to do and whether or not the node hit an error state, it's just logging success.

Reply:
True, I only went in half way (mostly because degradation under those conditions was more the thing I was worried about, not necessarily the doing nothing).

If I check the before and after to verify that it did nothing across kernel/kernelargs/extensions, is that acceptable?

Reply:
sure, adding a check to verifying what you consider to be success (for the non-coreos node case) in the else would wfm

Reply:
I think modifying current e2e gcp-op test doesn't makes sense today. Tests are written and added as we go on adding additional coverage. I don't know if it is possible today to create  a OCP cluster in ci with mixed nodes(RHCOS + RHEL) in a pool. Before improving these tests it would be worthwhile checking it.